api/auth.mjs

```diff
@@ -45,15 +45,20 @@ GET.auth.discord = async ({ url }) => {
   const { id: discordId } = await userResponse.json()
   const user = { ...session.user, discordId }
   const pendingUpdate = db.set(session.name, user)
+
+  // check  if member already in build
+  const guildMemberRequest = await fetch(`${DISCORD}/guilds/${GUILD}/members`);
+  const guildMembers = await guildMemberRequest.json();
+  const isMemberAlreadyInGuild = guildMembers.find(m => m.user.id ==== discordId);
   // join discord server
-  const join = await fetch(`${DISCORD}/guilds/${GUILD}/members/${discordId}`, {
-    method: 'PUT',
+  const join =  await fetch(`${DISCORD}/guilds/${GUILD}/members/${discordId}`, {
+    method: isMemberAlreadyInGuild ? 'PATCH' : "PUT",
```

Comment:
here i change the method , PATCH if variable isMemberAlreadyInGuild is true and PUT isn't